api/test/integration/test_agent_POST_endpoints.tavern.yaml

```diff
@@ -1,19 +1,109 @@
 ---
-test_name: POST /agents
+test_name: POST /agents (IPV6)
 
 marks:
   - base_tests
 
 stages:
 
     # POST /agents
-  - name: Try to create a new agent without body
+  - name: Create a new agent with a valid IPV6
     request: &agent_request
       verify: False
       url: "{protocol:s}://{host:s}:{port:d}/agents"
       method: POST
       headers:
         Authorization: "Bearer {test_login_token}"
+      json:
+        name: "NewIPV6Agent"
+        ip: "fdb9:8d4a:1fc2:12a0:11aa:22ff:33eb:99cd"
+    response:
+      status_code: 200
+      json: &agent_create_response
+        error: !anyint
+        data:
+          id: !anystr
+          key: !anystr
+      save:
+        json:
+          agent1_id_ipv6: data.id
+
+    # POST /agents
+  - name: Create a new agent with a compressed IPV6
+    request:
+      verify: False
+      <<: *agent_request
+      json:
+        name: "NewCompressedIPV6Agent"
+        ip: "fdb9::99cd"
+    response:
+      status_code: 200
+      json:
+        <<: *agent_create_response
+      save:
+        json:
+          agent2_id_ipv6: data.id
+
+    # POST /agents
+  - name: Create a new agent with an invalid IPV6
+    request:
+      verify: False
+      <<: *agent_request
+      json:
+        name: "InvalidIPV6Agent"
+        ip: "fdb9:8d4a:1fc2:12a0:11aa:22ff:33eb:99gh"
+    response:
+      status_code: 400
+      json:
+        error: 1706
+
+    # POST /agents
+  - name: Create a new agent with an invalid compressed IPV6
+    request:
+      verify: False
+      <<: *agent_request
+      json:
+        name: "InvalidIPV6Agent"
+        ip: "fdb9::99gh"
+    response:
+      status_code: 400
+      json:
+        error: 1706
+    delay_after: !float "{global_db_delay}"
+
+    # GET /agents
+  - name: Check all registered IPV6 agents
+    request: &agent_list_response
+      verify: False
+      url: "{protocol:s}://{host:s}:{port:d}/agents"
+      method: GET
+      headers:
+        Authorization: "Bearer {test_login_token}"
+      params:
+        agents_list: "{agent1_id_ipv6:s},{agent2_id_ipv6:s}"
+    response:
+      status_code: 200
+      json:
+        error: !anyint
+        data:
+          affected_items:
```

Comment:
I think that we could also save the IPs and check that they are in this response.

Reply:
After we discussed this, we agreed that testing if the IP was correctly added won't be verified since it's not a part of our test scope. We should be testing that we can use the API for agent registration with no more further controls.